inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -0,0 +1,170 @@
+"""Protein solvation scenario."""
+from typing import Optional
+import os
+import shutil
+from inductiva.types import Path
+from inductiva.molecules.simulators import GROMACS
+from inductiva.utils.files import resolve_path, get_timestamped_path
+from inductiva.utils.misc import split_camel_case
+from inductiva.utils.templates import (TEMPLATES_PATH,
+                                       batch_replace_params_in_template)
+from inductiva.utils.files import remove_files_with_tag
+
+SCENARIO_TEMPLATE_DIR = os.path.join(TEMPLATES_PATH, "protein_solvation")
+GROMACS_TEMPLATE_INPUT_DIR = "gromacs"
+
+
+class ProteinSolvation():
+    """Solvated protein scenario."""
+
+    def __init__(self):
+        """Scenario constructor for protein solvation based on the 
+        GROMACS simulator.
+        The three main steps of this scenario are solvation, 
+        energy minimization and simulation. The user can control 
+        the number of steps used to perform the energy minimization step and 
+        the duration, temperature and integrator used to perform 
+        the simulation.
+        """
+        self.template_dir = os.path.join(SCENARIO_TEMPLATE_DIR,
+                                         GROMACS_TEMPLATE_INPUT_DIR)
+        self.simulator = GROMACS()
+
+    def simulate(
+            self,
+            protein_pdb: str,
+            working_dir: Optional[Path] = None,
+            simulation_time: float = 10,  # ns
+            temperature: float = 300,
```

Comment:
Shouldn't temperature and the protein be part of the scenario instead of the simulation? I'm asking because e.g. in the FluidBlock scenario, stuff that is related with the physical properties of the scenario are defined in the constructor of the class (kinematic viscosity, ...) and only properties related with the simulation are specified in the simulate method.

Reply:
Yes, also the protein can be part of the constructor of the scenario.....

Reply:
I was undecided about what parameters put on the constructor and what to put on the simulate. For example when I am constructing the dataset I define the temperature and simulation steps and then simulate this scenario for different proteins. That's why i put the protein in the simulate. But do you think that the scenario should be specific to the protein? I think that temperature should stay in the simulate method, since you can have the same protein simulated at different temperatures.